aiida_vasp/parsers/vasp.py

```diff
@@ -198,22 +231,58 @@ def missing_critical_file():
 
         return self.result(success=True)
 
-    def _set_parsable_quantities(self):
-        """Set the parsable_quantities dictionary based on parsable_items obtained from the FileParsers."""
-
-        import copy
+    def _check_priority_and_add(self):
+        """Add a parser based on priority and availability of the files to be parsed."""
 
         # Gather all parsable items as defined in the file parsers.
+        # We introduce the concept of priority which can be defined in
+        # FILE_PARSER_SETS as the key 'priority' and a number. If a
+        # quantity is present in several different parsers, the parser
+        # with the highest number will persist. The file also has to exist.
+        # It is still fully possible to create custom parser and/or utilize the
+        # alternatives approach.
+        priority = {}
         for filename, value in self._parsers.iteritems():
             # initialise the instance of this FileParser to None.
-            if filename in self._parsable_quantities:
-                raise RuntimeError('The quantity {0} has been defined by two FileParser classes.'.format(filename) +
-                                   ' Quantity names must be unique. If both quantities are equivalent, define one as' +
-                                   ' an alternative for the other.')
             value.parser = None
             for quantity, quantity_dict in value['parser_class'].PARSABLE_ITEMS.iteritems():
-                # Create quantity objects.
-                self.add_parsable_quantity(quantity, quantity_dict, self.out_folder.get_folder_list())
+                add = False
+                if value.priority is not None:
```

Comment:
Consider splitting the inner loop off into a separate function to increase readability.

Reply:
Removed from the public repo. But refactored anyway.